pallet/src/migrations.rs

```diff
@@ -0,0 +1,81 @@
+use super::*;
+
+use frame_support::{
+    dispatch::GetStorageVersion, pallet_prelude::StorageVersion, traits::Get, weights::Weight,
+};
+use sp_std::vec::Vec;
+use structs::*;
+
+pub(crate) fn on_runtime_upgrade<T: Config>() -> Weight {
+    MigrateToV1x::<T>::on_runtime_upgrade()
+}
+
+// This migration sort all values in some StorageMaps allowing them to be binary searchable
+// As there was no storage version set before, we set it now to 0
+pub struct MigrateToV1x<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> MigrateToV1x<T> {
+    pub fn on_runtime_upgrade() -> Weight {
+        let target_storage_version: StorageVersion = Pallet::<T>::current_storage_version();
+        let on_chain_storage_version: StorageVersion = Pallet::<T>::on_chain_storage_version();
+
+        let mut weight: u64 = 0;
+        if on_chain_storage_version < target_storage_version {
+            log::info!(
+                "Pallet RBAC: Migration from onchain version {:?} to current version {:?}",
+                on_chain_storage_version,
+                target_storage_version,
+            );
+
+            // TODO repetitive translation logic with different types, reduce this with a macro
+            // translate all vec to sorted vec
+            Role2UserStore::<T>::translate_values::<Vec<Role2User<T::EntityId>>, _>(
```

Comment:
I feel this solution will encounter the scalability issue because the storage may be huge and reach the weight limit during the upgrade. However, it should work fine now because Agung/Krest/Peaq didn't have many entries. Let us double-check on the runtime upgrade test

Next time, we should think about the on-the-fly storage migration.

Reply:
I see the point, Although I didn't come across a more optimal solution. I'll check it against the runtime upgrade test.

Reply:
We will work on this in the future, for now it will do.